### PR TITLE
Command Completion

### PR DESCRIPTION
**Pull request by IdlePhysicist**
I have added command completion for bash and zsh shells.

Comments welcome.

### CODE REVIEW

File: cmd/completion/completion.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+		Args: cobra.ExactValidArgs(1),
+
+		Run: func(cmd *cobra.Command, args []string) {
+			switch args[0] {
+			case `bash`:
+				rootCmd.GenBashCompletion(os.Stdout)
+			case `zsh`:
+				rootCmd.GenZshCompletion(os.Stdout)
+				io.WriteString(os.Stdout, "\ncompdef _knut knut\n")
```
Comment (sboehler) [inline cmd/completion/completion.go:39]: What does this line do? I don't see it in https://github.com/spf13/cobra/blob/master/shell_completions.md

**Review by sboehler — CHANGES_REQUESTED** [review]
Thanks, that is a great feature! I tested your version for bash, and it seems to work.

Can we just copy the file from here (https://github.com/spf13/cobra/blob/master/shell_completions.md), to get the full help text? I think that could be useful!

File: cmd/completion/completion.go (hunk 1 above)
Comment (IdlePhysicist) [reply cmd/completion/completion.go:62]: That is an interesting line it sources the completion definition file, and tells ZSH what command the completions are for [1]. Cobra does not include this line in the generated output [2]. 

[1] https://github.com/zsh-users/zsh-completions/blob/master/zsh-completions-howto.org#telling-zsh-which-function-to-use-for-completing-a-command

[2] https://github.com/helm/helm/blob/master/cmd/helm/completion.go#L167

Comment (sboehler) [reply cmd/completion/completion.go:65]: Thanks!

**Pull request merged**